src/optimizer/plugins/RemoveGeneratedTextRulesOptimizer.ts

```diff
@@ -1,36 +1,69 @@
+import { remove } from 'lodash';
 import { OptimizerPlugin } from '../OptimizerPlugin';
 import { Package } from '../../processor';
-import { ExportableCaretValueRule, ExportableRule } from '../../exportable';
-import { pullAt } from 'lodash';
+import {
+  ExportableAssignmentRule,
+  ExportableCaretValueRule,
+  ExportableCodeSystem,
+  ExportableExtension,
+  ExportableInstance,
+  ExportableProfile,
+  ExportableRule,
+  ExportableValueSet
+} from '../../exportable';
 import { fshtypes } from 'fsh-sushi';
 
 export default {
   name: 'remove_generated_text_rules',
-  description: 'Remove caret rules on "text" which represent generated content.',
+  description: 'Remove rules on "text" which represent generated content.',
 
   optimize(pkg: Package): void {
     [...pkg.profiles, ...pkg.extensions, ...pkg.valueSets, ...pkg.codeSystems].forEach(resource => {
-      const rulesToRemove: number[] = [];
-      const hasGeneratedText = resource.rules.some(
-        (rule: ExportableRule) =>
-          rule instanceof ExportableCaretValueRule &&
-          rule.path === '' &&
-          rule.caretPath === 'text.status' &&
-          rule.value instanceof fshtypes.FshCode &&
-          ['generated', 'extensions'].includes(rule.value.code)
-      );
-      if (hasGeneratedText) {
-        resource.rules.forEach((rule: ExportableRule, i: number) => {
-          if (
+      if (hasGeneratedText(resource)) {
+        remove(
+          resource.rules as ExportableRule[],
+          rule =>
             rule instanceof ExportableCaretValueRule &&
             rule.path === '' &&
             rule.caretPath.match(/^text\./)
-          ) {
-            rulesToRemove.push(i);
-          }
-        });
+        );
+      }
+    });
+
+    [...pkg.instances].forEach(instance => {
+      if (hasGeneratedText(instance)) {
+        instance.rules = instance.rules.filter(
+          rule => !(rule instanceof ExportableAssignmentRule && rule.path.match(/^text\./))
+        );
```

Comment:
Maybe this is pedantic, but for consistency, it seems like since you used `remove` above, you should use `remove` here, since it's kind of a parallel situation.

Reply:
https://github.com/FHIR/GoFSH/pull/45/commits/7ca90a984381e68d80c737b4acebd0347172a358